docs/source/conf.py

```diff
@@ -26,6 +30,62 @@
 autodoc_member_order = "bysource"
 autodoc_typehints = "signature"
 
+
+# -- sphinx.ext.linkcode
+def linkcode_resolve(domain: str, info: dict) -> Optional[str]:
+    """
+    Returns the URL to the GitHub source code corresponding to a object,
+    if identified. Otherwise returns None.
+
+    Parameters
+    ----------
+    domain: str
+        The domain of the object (e.g., "py" for Python, "cpp" for C++).
+    info: dict
+        Dictionary containing the module and fullname of the object.
+
+    Returns
+    -------
+    Optional[str]
+        The URL to the GitHub source code corresponding to the object.
+        If a URL cannot be generated, returns None instead.
+    """
+    if domain != "py" or not info.get("module") or not info.get("fullname"):
+        return None
+
+    # Find the object.
+    module = importlib.import_module(info["module"])
+
+    if "." not in info["fullname"]:  # object is a class or function
+        obj = getattr(module, info["fullname"])
+    else:  # object is an attribute of a class
+        obj_name, attr_name = info["fullname"].split(".")
+        obj = getattr(module, obj_name)
+        obj = getattr(obj, attr_name)
+
+    # Find the object's source file and starting line number.
+    try:
+        source_file = inspect.getsourcefile(obj)
+        assert source_file is not None, "Source file not found."
+    except (TypeError, AssertionError):
+        return None
+
+    parent_dir_path = os.path.abspath("..")
+    rel_path = os.path.relpath(source_file, parent_dir_path)
+    start_line_no = inspect.getsourcelines(obj)[1]
+
+    # Get the current git revision. If it cannot be determined, return "main".
+    try:
+        cmd = "git rev-parse --short HEAD".split()
+        output = subprocess.check_output(cmd).strip()
+        revision = output.decode("utf-8")
+    except (subprocess.CalledProcessError, OSError):
+        revision = "main"
```

Comment:
Doesn't work yet for RTD.

Reply:
Maybe need to use [sphinx-git](https://sphinx-git.readthedocs.io/en/latest/) for this.

Reply:
https://readthedocs.org/api/v2/build/21846816.txt

CTRL-f "git revision" shows that the function does identify the correct git revision. But why then do the links not get resolved correctly?

Reply:
Found the problem: the URLs are wrong 😅 I ran `poetry run make docs` from inside `docs/`, so the paths are different than when it's called from the root directory.

```
https:///github.com/PyVRP/PyVRP/blob/4bf717d/../pyvrp/plotting/plot_coordinates.py#L9
https:///github.com/PyVRP/PyVRP/blob/4bf717d/../pyvrp/plotting/plot_demands.py#L9
```